libraries/AP_AS5600_AOA/AS5600_AOA.cpp

```diff
@@ -0,0 +1,266 @@
+/****************************************************
+ * AMS 5600 Angle of Attack Class for Ardupilot platform
+ * Author: Cole Mero
+ * Date: 15 Dec 2014
+ * File: AS5600_AOA.cpp
+ * Version 1.00
+ * www.ams.com
+ *
+ * Description:  This class has been designed to
+ * access the AS5600 magnetic encoder sensor to
+ * read the angle of attack and record it for experimental purposes
+ *
+ * It was adapted from the Arduino library available for the AS5600 sensor.
+ *
+***************************************************/
+
+#include <unistd.h>
+#include "AS5600_AOA.h"
+
+extern const AP_HAL::HAL &hal;
+
+/****************************************************
+ * Method: AS5600_AOA
+ * In: none
+ * Out: none
+ * Description: constructor for AMS 5600
+***************************************************/
+ AS5600_AOA::AS5600_AOA()
+{
+
+    bus = 1; //Sets the bus number for the device, unclear what this number should be, trial and error to make it work
+    address = 0x36; //This is the I2C address for the device, it is set by the manufacturer
+
+   /*load register values*/
+
+  _zmco = 0x00;
+  _zpos_hi = 0x01;
+  _zpos_lo = 0x02;
+  _mpos_hi = 0x03;
+  _mpos_lo = 0x04;
+  _mang_hi = 0x05;
+  _mang_lo = 0x06;
+  _conf_hi = 0x07;
+  _conf_lo = 0x08;
+  _raw_ang_hi = 0x0c;
+  _raw_ang_lo = 0x0d;
+  _ang_hi = 0x0e;
+  _ang_lo = 0x0f;
+  _stat = 0x0b;
+  _agc = 0x1a;
+  _mag_hi = 0x1b;
+  _mag_lo = 0x1c;
+  _burn = 0xff;
+
+}
+
+
+void AS5600_AOA::init(){
+
+    dev = hal.i2c_mgr->get_device(bus, address);
+
+    WITH_SEMAPHORE(dev->get_semaphore());
+    dev->set_speed(AP_HAL::Device::SPEED_LOW);
+    dev->set_retries(2);
+
+}
+
+/* mode = 0, output PWM, mode = 1 output analog (full range from 0% to 100% between GND and VDD*/
+void AS5600_AOA::setOutPut(unsigned char mode){

```

Comment:
```suggestion
void AS5600_AOA::setOutPut(uint8_t mode)
{

```

Need to use stdint types throughout.  I haven't pointed all these places out.